### ISSUE TITLE
Consider adding a conan.lock for stability

### ISSUE THREAD

**Issue by memsharded** [carimbolabs/carimbo#1]
Hi there!

Depending on the needs, and who is building and using the engine, it might be beneficial to add a ``conan.lock`` in the repo itself. This way, your CI, your users, etc, all use exactly the same verisons and revisions of the dependencies, minimizing possible issues when ConanCenter packages evolve (they do, and quite often, even if there is CI, it is always possible that new versions might break something)

The ``conan.lock`` can be committed in the repo, and it is picked automatically by Conan 2 ``conan install`` if present, so no need of extra args.

**Comment by skhaz** [comment]
Thank you, nice tip! I will do it.

**Comment by skhaz** [comment]
@memsharded One issue I encountered when using the lock: I generated the lock on macOS, and when building the project on Ubuntu, an Ubuntu-specific dependency was missing from the lockfile.

```
ERROR: Requirement 'openssl/[>=1.1 <4]' not in lockfile 'requires'
```

**Comment by memsharded** [comment]
Yes, when a lockfile already exists, it will fail if some dependencies are not in the lockfile (strict-checking).

It can be relaxed with:
```
conan install ... --lockfile-partial
```
And ``conan lock create`` doesn't need that argument, it is assumed by default.
A possible way to capture a lockfile that covers all the different configurations is:
```
conan lock create . -pr=profile
conan lock create . -pr=profile2
...
```
The lockfile will accumulate the different dependencies in the different profiles (systems). As ``conan lock create`` doesn't really install the binaries, it can be executed in the same machine even for other platforms, to get a single common lockfile.

**Comment by skhaz** [comment]
Sounds good, thank you!
